### PR TITLE
remove numpy.core warning filter, link to issue on pyarrow caused BlockManager warning [test-upstream]

### PR DESCRIPTION
**Pull request by graingert**
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -148,9 +148,9 @@ filterwarnings = [
     "ignore:'H' is deprecated and will be removed in a future version. Please use 'h' instead of 'H':FutureWarning",
     'ignore:DataFrameGroupBy\.apply operated on the grouping columns\. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation\. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning\.:FutureWarning',
     "ignore:'BA' is deprecated and will be removed in a future version. Please use 'BY' instead of 'BA':FutureWarning",
-    'ignore:numpy\.core.* is deprecated and has been renamed to numpy\._core\.*:DeprecationWarning', # needed for pandas to import
+    'ignore:numpy\.core.* is deprecated and has been renamed to numpy\._core\.*:DeprecationWarning:pandas', # needed for pandas to import
     'ignore:Passing a BlockManager to DataFrame is deprecated and will raise in a future version. Use public APIs instead:DeprecationWarning',
```
Comment (graingert) [inline pyproject.toml:152]:
```suggestion
    'ignore:Passing a BlockManager to DataFrame is deprecated and will raise in a future version. Use public APIs instead:DeprecationWarning:pyarrow',
```

File: pyproject.toml [hunk 2]
```diff
@@ -148,8 +148,8 @@ filterwarnings = [
     "ignore:'H' is deprecated and will be removed in a future version. Please use 'h' instead of 'H':FutureWarning",
     'ignore:DataFrameGroupBy\.apply operated on the grouping columns\. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation\. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning\.:FutureWarning',
     "ignore:'BA' is deprecated and will be removed in a future version. Please use 'BY' instead of 'BA':FutureWarning",
-    'ignore:numpy\.core.* is deprecated and has been renamed to numpy\._core\.*:DeprecationWarning', # needed for pandas to import
-    'ignore:Passing a BlockManager to DataFrame is deprecated and will raise in a future version. Use public APIs instead:DeprecationWarning',
+    'ignore:numpy\.core.* is deprecated and has been renamed to numpy\._core\.*:DeprecationWarning:pandas', # needed for pandas to import
```
Comment (graingert) [inline pyproject.toml:151]: this is fixed now https://github.com/pandas-dev/pandas/pull/55565/files
```suggestion
```

**Comment by graingert** [comment]
we're using numpy.core distributed fixes here https://github.com/dask/distributed/pull/8291/files

File: pyproject.toml [hunk 3]
```diff
@@ -141,8 +141,8 @@ filterwarnings = [
     "ignore:.*:dask.tests.warning_aliases.RemovedIn20Warning",
     "ignore:When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas:FutureWarning",
     'ignore:DataFrameGroupBy\.apply operated on the grouping columns\. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation\. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning\.:FutureWarning',
+    'ignore:Passing a BlockManager to DataFrame is deprecated and will raise in a future version. Use public APIs instead:DeprecationWarning',  # https://github.com/apache/arrow/issues/35081
     'ignore:numpy\.core.* is deprecated and has been renamed to numpy\._core\.*:DeprecationWarning', # needed for pandas to import
```
Comment (graingert) [inline pyproject.toml:145]:
```suggestion
```

**Review by hendrikmakait — APPROVED** [review]

**Pull request merged**